src/app/shared/infinity-scroll/infinity-scroll.directive.ts

```diff
@@ -0,0 +1,28 @@
+import {Directive, output} from '@angular/core';
+
+@Directive({
+    selector: '[appScrollWithLoading]',
+    standalone: true,
+    host: {
+        '(scroll)': 'onScroll($event)',
+    },
+})
+export class InfinityScrollDirective {
+    private lastScrollTop = 0;
+    loadNextData = output<void>();
+
+    onScroll(event: Event) {
+        const element = event.target as HTMLElement;
+        const scrollTop = element.scrollTop;
+        const clientHeight = element.clientHeight;
+        const scrollHeight = element.scrollHeight;
```

Comment:
Рекомендую воспользоваться деструктуризацией, супер удобная и читаемая вещь
```suggestion
        const {scrollTop, clientHeight, scrollHeight} = element;
```